creator-node/src/fileManager.js

```diff
@@ -95,57 +96,105 @@ async function saveFileToIPFSFromFS (req, srcPath, fileType, sourceFile, transac
 
 /** Save file to disk given IPFS multihash, and ensure availability.
  *  Steps:
- *  - If file already stored on disk, return immediately.
- *  - If file not already stored, fetch from IPFS and store.
+ *  - If file already stored on disk, return immediately and store to disk.
+ *  - If file not already stored, fetch from IPFS and store to disk.
  *    - If multihash available on local inode, retrieve file.
  *    - If multihash not available locally, fetch file from IPFS.
- *  - Write file to disk.
+ *  - If file is not available via IPFS try other cnode gateways for user's replica set.
  *  - Add file to local inode if not already.
  */
-async function saveFileForMultihash (req, multihash, expectedStoragePath) {
+async function saveFileForMultihash (req, multihash, expectedStoragePath, gatewaysToTry) {
   // If file already stored on disk, return immediately.
   if (fs.existsSync(expectedStoragePath)) {
     req.logger.info(`File already stored at ${expectedStoragePath} for ${multihash}`)
     return expectedStoragePath
   }
 
   // If file not already stored, fetch from IPFS and store at storagePath.
-  const ipfs = req.app.get('ipfsAPI')
   let fileBuffer = null
-  req.logger.info(`Storing file at ${expectedStoragePath} for track multihash ${multihash}`)
 
   // If multihash already available on local INode, cat file from local ipfs node
-  req.logger.info(`checking if ${multihash} already available on local ipfs node`)
+  req.logger.debug(`checking if ${multihash} already available on local ipfs node`)
   try {
-    fileBuffer = await ipfs.cat(multihash)
-    req.logger.info(`Retrieved file for ${multihash} from local ipfs node`)
+    fileBuffer = await Utils.ipfsCat(multihash, req, 500)
+    req.logger.debug(`Retrieved file for ${multihash} from local ipfs node`)
+    // Write file to disk.
+    await writeFile(expectedStoragePath, fileBuffer)
+    req.logger.info(`wrote file to ${expectedStoragePath}, obtained via ipfs cat`)
   } catch (e) {
     req.logger.info(`Multihash ${multihash} is not available on local ipfs node`)
   }
 
   // If file not already available on local INode, fetch from IPFS.
   if (fileBuffer === null) {
-    req.logger.info(`Attempting to get ${multihash} from IPFS`)
+    req.logger.debug(`Attempting to get ${multihash} from IPFS`)
     let output
     try {
-      output = await ipfs.get(multihash)
+      // ipfsGet returns a BufferList object, not a buffer
+      // but still compatible into writeFile directly
+      output = await Utils.ipfsGet(multihash, req, 5000)
+      fileBuffer = output
+      req.logger.debug(`retrieved file for multihash ${multihash} from local ipfs node`)
+
+      // Write file to disk.      
+      const destinationStream = fs.createWriteStream(expectedStoragePath)
```

Comment:
if the output is still compatible with writeFile, why not use it instead of createWriteStream?

Reply:
it's not compatible with writeFile since this is a stream and writeFile only supports buffers

Reply:
was going off your comment on line 134 "still compatible into writeFile directly"